x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java

```diff
@@ -529,6 +543,9 @@ private void updateTargetIndexSettingStep(
                 updateSettingsReq,
                 new UpdateDownsampleIndexSettingsActionListener(listener, parentTask, downsampleIndexName, request.getWaitTimeout())
             );
+
+        // Record latency for downsampling operation.
+        downsampleMetrics.recordLatencyMaster(getDurationInMillis(), DownsampleMetrics.ShardActionStatus.SUCCESS);
```

Comment:
IMO this is more well placed in the caller where we log "All downsampling tasks completed".

Reply:
Indeed, done.